File: 2021/05/10/main.py
class Solution:
    def countPrimes(self, n: int) -> int:
        primes = [True for _ in range(n+1)]
        i = 2
        for i in range(2, n+1):
            if primes[i] == False:
                continue
            for j in range(i*i, n, i):
                primes[j] = False
        r = 0
        for i in range(2, n):
            if primes[i]:
                r += 1
        return r

File: 2021/05/10/main.go
package main

func countPrimes(n int) int {
	primes := make([]bool, n+1)
	for i := 2; i < n; i++ {
		primes[i] = true
	}

	for i := 2; i*i <= n; i++ {
		if !primes[i] {
			continue
		}
		for j := i * i; j < n; j += i {
			primes[j] = false
		}
	}
	r := 0
	for i := 2; i < n; i++ {
		if primes[i] {
			r++
		}
	}
	return r
}
